bats/index.bats

```diff
@@ -1993,6 +1993,30 @@ SQL
     [ "$status" -eq "0" ]
 }
 
+@test "index: UNIQUE allows multiple NULL values" {
+    dolt sql <<SQL
+CREATE TABLE test (
+  pk BIGINT PRIMARY KEY,
+  v1 BIGINT,
+  UNIQUE INDEX (v1)
+);
+CREATE TABLE test2 (
+  pk BIGINT PRIMARY KEY,
+  v1 BIGINT,
+  v2 BIGINT,
+  UNIQUE INDEX (v1, v2)
+);
+INSERT INTO test VALUES (0, NULL), (1, NULL), (2, NULL);
+INSERT INTO test2 VALUES (0, NULL, NULL), (1, NULL, NULL), (2, 1, NULL), (3, 1, NULL), (4, NULL, 1), (5, NULL, 1);
+SQL
+    run dolt sql -q "SELECT * FROM test" -r=json
+    [ "$status" -eq "0" ]
+    [[ "$output" =~ '{"rows": [{"pk":0},{"pk":1},{"pk":2}]}' ]] || false
+    run dolt sql -q "SELECT * FROM test2" -r=json
+    [ "$status" -eq "0" ]
+    [[ "$output" =~ '{"rows": [{"pk":0},{"pk":1},{"pk":2,"v1":1},{"pk":3,"v1":1},{"pk":4,"v2":1},{"pk":5,"v2":1}]}' ]] || false

```

Comment:
Need a test here exercising the index, e.g. where clause that references the indexed column, including a case for = NULL